displaylink fix: ignore primary argument when using evdi drivers

nktnet1 opened the pull request:
Fixes #22.

A slightly more robust way to address the issue than the hack provided in #22 - done by checking if the driver being used for the given GPU is `evdi`. If so, set `primary` to `{}`.

Unsure of the following:
- whether this still falls under "hacky code" mentioned in Hyprland's [PR Guidelines](https://wiki.hyprland.org/Contributing-and-Debugging/PR-Guidelines/#pr-requirements)
- variables like `drmName` and `drmVer` are `auto` instead of `auto const` - have followed the same convention for the newly introduced `drmVerName` variable
- whether there exist use cases with `evdi` drivers and using the passed-in `primary_` argument (I assume no)

vaxerski commented:
hm, I am no kernel dev, but this will essentially force KMS to take buffers from your primary gpu instead of them living on evdi.

I have no clue if that has any ramifications for the user.

If it works fine and others can attest to that, I'm fine with this.

vaxerski commented on src/backend/drm/DRM.cpp:
```diff
@@ -599,8 +599,13 @@ bool Aquamarine::CDRMBackend::registerGPU(SP<CSessionDevice> gpu_, SP<CDRMBacken
     gpuName = drmName;
 
+    auto drmVerName = drmVer->name ? drmVer->name : "unknown";
+    if (std::string_view(drmVerName) == "evdi") {
```
no {}

nktnet1 commented:
Thanks @vaxerski - I've fixed the style issue. Running clang-format also modified `tests/SimpleWindow.cpp` - happy to revert if that should be done in a different MR.

> I have no clue if that has any ramifications for the user.

I'm uncertain of this too - my assumption based on what I've read and researched thus far is that evdi is incapable of rendering:
![image](https://github.com/user-attachments/assets/e6894de8-56ec-43c4-a5ed-bd5177993415)


> If it works fine and others can attest to that, I'm fine with this.

I think it's working fairly well for those on wlroots:
![evidence-users](https://github.com/user-attachments/assets/f947f3c7-f042-4e10-8acf-e076944026c4)

and here's a reply to the previous version of my patch (although effectively the same changes):
- https://github.com/hyprwm/Hyprland/issues/2752#issuecomment-2242770323

nktnet1 commented:
Adding to findings above, I've also tested this patch with the nvidia-gpu re-enabled

Here are the logs:
- [patched-with-nvidia.log](https://github.com/user-attachments/files/16362419/patched-with-nvidia.log)

`lspci` and `/dev/dri/by-path`:
![image](https://github.com/user-attachments/assets/7085ca27-554d-45ee-944d-8a4727efb917)

<details close>
<summary>Text output</summary>

```sh
~$ lspci | grep -E 'VGA|3D'
0000:00:02.0 VGA compatible controller: Intel Corporation Raptor Lake-S UHD Graphics (rev 04)
0000:01:00.0 VGA compatible controller: NVIDIA Corporation AD107M [GeForce RTX 4060 Max-Q / Mobile] (rev a1)
~$ ls -l /dev/dri/by-path  
total 0
lrwxrwxrwx 1 root root  8 Jul 24 22:55 pci-0000:00:02.0-card -> ../card1
lrwxrwxrwx 1 root root 13 Jul 24 22:55 pci-0000:00:02.0-render -> ../renderD128
lrwxrwxrwx 1 root root  8 Jul 24 22:55 pci-0000:01:00.0-card -> ../card0
lrwxrwxrwx 1 root root 13 Jul 24 22:55 pci-0000:01:00.0-render -> ../renderD129
lrwxrwxrwx 1 root root  8 Jul 24 22:55 platform-evdi.0-card -> ../card2
lrwxrwxrwx 1 root root  8 Jul 24 22:55 platform-evdi.1-card -> ../card3
~$ 
```

</details>

vaxerski commented:
> Running clang-format also modified tests/SimpleWindow.cpp - happy to revert if that should be done in a different MR

ye please do, I should remove it anyways its broken

nktnet1 commented:
> > Running clang-format also modified tests/SimpleWindow.cpp - happy to revert if that should be done in a different MR
> 
> ye please do, I should remove it anyways its broken

Have been reverted :)
